### PR TITLE
Team 1 : Testing of JPA and modification in entities.

### PR DESCRIPTION
**Pull request by Haseeb08**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -68,6 +72,37 @@
 			<artifactId>spring-security-test</artifactId>
 			<scope>test</scope>
 		</dependency>
+		<dependency>
```
Comment (ghost) [inline pom.xml]: Why did you add junit separately? That too twice?
Junit is already part of spring test.

File: pom.xml [hunk 2]
```diff
@@ -68,6 +72,37 @@
...
+			<artifactId>spring-test</artifactId>
+			<version>4.3.7.RELEASE</version>
+		</dependency>
+
+		<dependency>
+			<groupId>com.h2database</groupId>
+			<artifactId>h2</artifactId>
+			<version>1.4.194</version>
+		</dependency>
+		<dependency>
```
Comment (ghost) [inline pom.xml:86]: This is also already part of the Spring

**Review by ghost — CHANGES_REQUESTED** [review]
Ignore the previous comment of front end review

File: src/main/java/com/vikingzorros/rehabit/daorepositories/CommentRepository.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+import com.vikingzorros.rehabit.entities.Comment;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.List;
+
+public interface CommentRepository extends JpaRepository<Comment,Integer> {
+
+    List<Comment> findByPost_id(int theId);
```
Comment (ghost) [inline src/main/java/com/vikingzorros/rehabit/daorepositories/CommentRepository.java]: This is not java convention of naming

**Review by ghost — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (Haseeb08) [reply pom.xml:86]: Application is failing to start (while testing) without Junit and h2 dependency.

File: src/main/java/com/vikingzorros/rehabit/daorepositories/CommentRepository.java (hunk 3 above)
Comment (Haseeb08) [reply src/main/java/com/vikingzorros/rehabit/daorepositories/CommentRepository.java]: Changes done.

File: pom.xml (hunk 1 above)
Comment (Haseeb08) [reply pom.xml]: I deleted the repeated one.

File: src/main/java/com/vikingzorros/rehabit/daorepositories/PostRepository.java [hunk 4]
```diff
@@ -10,4 +10,5 @@
         List<Post> findByCategory_id(int theId);
        //  List<Post> findByCategory_idOrderByCreate_timeDesc(int theId);
 
+
```
Comment (shravanvemula) [inline src/main/java/com/vikingzorros/rehabit/daorepositories/PostRepository.java:13]: Method name is not according to the java method naming conventions

Comment (Haseeb08) [reply src/main/java/com/vikingzorros/rehabit/daorepositories/PostRepository.java:13]: I have changed that in last commit.

File: src/test/java/com/vikingzorros/rehabit/repository/UserRepositoryTest.java [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+@Slf4j
+@RunWith(SpringRunner.class)
+@DataJpaTest
+public class UserRepositoryTest {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private PostRepository postRepository;
```
Comment (shravanvemula) [inline src/test/java/com/vikingzorros/rehabit/repository/UserRepositoryTest.java]: You should not use PostRepository seperately here,try to fetch post from user object itself

Comment (Haseeb08) [reply src/test/java/com/vikingzorros/rehabit/repository/UserRepositoryTest.java]: Done

Comment (Haseeb08) [reply src/test/java/com/vikingzorros/rehabit/repository/UserRepositoryTest.java]: Changes done.

**Review by MallipeddiRuchitha — APPROVED** [review]

**Review by shravanvemula — APPROVED** [review]

**Review by shravanvemula — APPROVED** [review]
